Add swift-playground-mode

msanders opened the pull request:
### Brief summary of what the package does

 Emacs support for Swift playgrounds.

### Direct link to the package repository

https://gitlab.com/michael.sanders/swift-playground-mode

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

DamienCassou commented:
1. Please run linters specified in CONTRIBUTING.md. They would have told you about:

    - by default, 2 spaces must follow each dot to separate sentences
    - there are more than 1 argument to `pkg-info-version-info` in the call to `declare-function`
    - "non-nil" is usually preferred over "true" in docstrings
    - the footer references
    - you miss a `(require 'comint)` for `comint-check-proc`

2. as explained in the docstring of `read-only-mode`, it is recommended to temporarily bind `inhibit-read-only`
3. docstrings usually start with imperative verbs: "Close" instead of "Closes" in `close-buffer`
4. both `apply` and `funcall` can take many arguments, you don't have to put them all in a list first. This should work (untested):

    ```emacs
    (apply 'call-process
           (car command-list)
           infile
           destination
           display
           (cdr command-list)))    
    ```

5. `-call-process-with-output` says it returns the exit status but this seems wrong.
6. instead of `…-is-…`, predicate usually ends with `-p`: e.g., `current-buffer-playground-p`
7. usually, closing parenthesis are not on their own lines. You might want to read and follow  https://github.com/bbatsov/emacs-lisp-style-guide

DamienCassou commented:
I'm closing this for now. Feel free to reopen when ready.

msanders commented:
Sorry for the delay. I've updated this with the suggested changes.

Regarding the docstrings, I've updated `sentence-end-double-space` to be set to nil in a file-local variable, and `checkdoc` is passing now (which should be shown shortly on Travis).
